minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java

```diff
@@ -0,0 +1,426 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.multi.process;
+
+import alluxio.AlluxioTestDirectory;
+import alluxio.AlluxioURI;
+import alluxio.Configuration;
+import alluxio.ConfigurationRule;
+import alluxio.ConfigurationTestUtils;
+import alluxio.PropertyKey;
+import alluxio.cli.Format;
+import alluxio.client.file.FileSystem;
+import alluxio.client.file.FileSystem.Factory;
+import alluxio.client.file.FileSystemContext;
+import alluxio.exception.status.UnavailableException;
+import alluxio.master.LocalAlluxioCluster;
+import alluxio.master.MasterInquireClient;
+import alluxio.master.SingleMasterInquireClient;
+import alluxio.master.ZkMasterInquireClient;
+import alluxio.network.PortUtils;
+import alluxio.util.CommonUtils;
+import alluxio.util.network.NetworkAddressUtils;
+
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.io.Closer;
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.FileUtils;
+import org.apache.curator.test.TestingServer;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.ThreadLocalRandom;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Class for starting, stopping, and interacting with an Alluxio cluster where each master and
+ * worker runs in its own process.
+ *
+ * Compared to {@link LocalAlluxioCluster}, {@link MultiProcessCluster} is
```

Comment:
shall we actually rename `LocalAlluxioCluster` to `SingleProcessCluster` then? leave this up to you (and can be in different PRs)

Reply:
This touches a lot of code, so I'd prefer to do it in a separate PR after merging `external-minicluster` to master

Reply:
sg